benchbuild/project.py

```diff
@@ -237,6 +240,16 @@ def __default_primary_source(self) -> str:
     def __attrs_post_init__(self) -> None:
         db.persist_project(self)
 
+        # select container image
+        if isinstance(type(self).CONTAINER, ContainerImage):
+            self.container = type(self).CONTAINER
+        else:
+            version = self.version_of_primary
```

Comment:
I would like to have a check here, whether the primary source is based on a git repo, as this is required for the container selection to make any sense.
However, `self.primary_source` could have wrappers around it at this point (e.g., version filters).
I could check for `type(self).CONTAINER[0]`, but that would break our primary source abstraction.

Any suggestions?